### PR TITLE
chore: switch to testing only Rust shim in integration tests

### PR DESCRIPTION
**Pull request by chris-olszewski**
Now that we'll no longer be shipping a Go binary and only the Rust shim we should be testing against the shim exclusively.

This is also desirable since we can avoid compiling Go twice (once to produce a library and once to produce a binary) when running these tests in CI.

### CODE REVIEW

File: cli/integration_tests/basic_monorepo/profile.t [hunk 1]
```diff
@@ -4,6 +4,6 @@ Setup
 
 Run build and record a trace
 Ignore output since we want to focus on testing the generated profile
-  $ ${TURBO} build --profile=build.trace > turbo.log
```
Comment (NicholasLYang) [inline cli/integration_tests/basic_monorepo/profile.t:7]: I was thinking we just rename the `SHIM` variable to `TURBO` and then remove the go binary entirely.

**Review by NicholasLYang — COMMENTED** [review]
Nice yeah I was doing this in #2733 but it makes sense that we do it here. We could also port the e2e tests and package tests too.

**Comment by chris-olszewski** [comment]
Failures on the JS Package Tests should go away once this is merged: https://github.com/vercel/turbo/pull/2568

**Review by NicholasLYang — APPROVED** [review]

**Comment by chris-olszewski** [comment]
Ugh, before we land this we should make sure we only build the Rust shim once for each platform. The way things are setup now we're building it for e2e, integration, *and* JS package tests. This is way too many opportunities to timeout when downloading/compiling the Rust :(

**Comment by chris-olszewski** [comment]
Closing in favor of #2854 since this change by itself will add ~10 min to every PR that needs integration testing

**Pull request closed without merging**